sys/posix/include/semaphore.h

```diff
@@ -34,6 +34,9 @@
 extern "C" {
 #endif
 
+/**
+ * @brief A sema_t is a sem_t is a sema_t
```

Comment:
As far as I see `sys_sem_t` is required for lwip only.
@miri, any reason against dropping `sema_t` comepltely?